corehq/apps/users/tasks.py

```diff
@@ -43,33 +44,42 @@ def tag_docs_as_deleted(cls, docs, deletion_id):
     cls.get_db().bulk_save(docs)
 
 
-@task(rate_limit=2, queue='background_queue', ignore_result=True,
-      default_retry_delay=5 * 60, max_retries=3)
-def tag_forms_as_deleted_rebuild_associated_cases(formlist, deletion_id,
+@task(rate_limit=2, queue='background_queue', ignore_result=True, acks_late=True)
+def tag_forms_as_deleted_rebuild_associated_cases(form_id_list, deletion_id,
                                                   deleted_cases=None):
-    """ Upon user deletion, mark associated forms as deleted and prep cases
+    """
+    Upon user deletion, mark associated forms as deleted and prep cases
     for a rebuild.
     - 2 saves/sec for cloudant slowness (rate_limit)
-    - retry in 5 min if failure occurs after (default_retry_delay)
-    - retry a total of 3 times
     """
     if deleted_cases is None:
         deleted_cases = set()
 
     cases_to_rebuild = set()
-    for form in formlist:
-        form['doc_type'] += DELETED_SUFFIX
-        form['-deletion_id'] = deletion_id
+    forms_to_check = get_docs(XFormInstance.get_db(), form_id_list)
```

Comment:
it's already [chunked in the caller](https://github.com/dimagi/commcare-hq/pull/7710/files#diff-16ce9f2eb3be0695a91eca75714bb3b7R1677) so it didn't seem necessary. i could if we thought this method might be called elsewhere though